inst/www/htmlwidgets.js

```diff
@@ -567,8 +592,115 @@
+          resizeHandler
+        );
+      }
+
+      // This is needed for the specific case of ioslides, which
+      // flips slides between display:none and display:block.
+      // Ideally we would not have to have ioslide-specific code
+      // here, but rather have ioslides raise a generic event,
+      // but the rmarkdown package just went to CRAN so the
+      // window to getting that fixed may be long.
+      if (window.addEventListener) {
+        // It's OK to limit this to window.addEventListener
+        // browsers because ioslides itself only supports
+        // such browsers.
+        on(document, "slideenter", resizeHandler);
+        on(document, "slideleave", resizeHandler);
+      }
+    }
+
+    var scriptData = document.querySelector("script[data-for='" + el.id + "'][type='application/json']");
+    if (scriptData) {
+      var data = JSON.parse(scriptData.textContent || scriptData.text);
+      // Resolve strings marked as javascript literals to objects
+      if (!(data.evals instanceof Array)) data.evals = [data.evals];
+      for (var k = 0; data.evals && k < data.evals.length; k++) {
+        window.HTMLWidgets.evaluateStringMember(data.x, data.evals[k]);
+      }
+      binding.renderValue(el, data.x, initResult, initialState);
+      evalAndRun(data.jsHooks.render, initResult, [el, data.x]);
+    }
+  }
+
+  // Initialize an htmlwidget for Displayr.  At this point its assets (JavaScript, CSS) ought to have
+  // finished loading, and its body elements should be within the document's DOM. 
+  //
+  // el_parent  An element that contains a single rendered widget within it.
+  //
+  // Returns { binding, element, instance }.
+  window.HTMLWidgets.instantiateDisplayrWidget = function(el_parent, initialState, stateChangedHook) {
+    var bindings = window.HTMLWidgets.widgets || [];
+    var matches = bindings.map(binding => ({binding: binding, found: binding.find(el_parent)})).filter(e => e.found.length);
+    if (matches.length === 0)
+      throw new Error('No element was found matching an installed binding.  Have you loaded all assets?');
+    const binding = matches[0].binding;
+    const el = matches[0].found[0];
+    if (HTMLWidgets.getInstance(el))
+      throw new Error('A widget has already been instantiated on this DOM element.  Don\'t reuse DOM elements for widgets.');
+    var sizeObj = initSizingInABlock(el);
+    initializeWidget(el, binding, sizeObj, initialState, stateChangedHook);
+    return {
+      binding: binding,
+      element: el,
+      instance: HTMLWidgets.getInstance(el)
+    };
+  }
+
+  // Stops a Displayr widget.  Base Htmlwidgets do not support this concept, presuming that
+  // the entire page will be closed.  Callers are expected to remove the original element
+  // from the DOM.  `widget` should be what was returned by instantiateDisplayrWidget().
+  window.HTMLWidgets.destroyDisplayrWidget = function(widget) {
+    if (widget.instance.destroy)
```

Comment:
I'm missing the part where we remove the widget from the window.HTMLWidgets.widgets collection..

Reply:
The HTMLWidgets.widgets collection contains registered widget _types_ (bindings) - not individual widgets.  There is no mechanism to unregister a widget binding - it is registered when its source file is parsed, and remains in case we need it again.